app/components/Settings/ConnectButton/ConnectButton.react.js

```diff
@@ -39,26 +94,20 @@ export default class ConnectButton extends Component {
             const genericErrorMessage = 'Hm... had trouble connecting.';
             const errorMessage = connectErrorMessage || saveErrorMessage || genericErrorMessage;
             error = <div className={'errorMessage'}>{errorMessage}</div>;
-        } else if (
-            isLoading(connectRequest.status) ||
-            isLoading(saveConnectionsRequest.status)
-        ) {
-            buttonText = 'Connecting...';
-        } else if (
-            connectRequest.status >= 200 &&
-            connectRequest.status < 300
-        ) {
-            if (editMode) {
-                buttonText = 'Save changes';
-                buttonClick = connect;
-            } else {
+
+        } else if (this.isConnected()) {
+            buttonText = 'Save changes';
```

Comment:
'`Save changes`' should be shown only after we have successfully connected to a DB (i.e. `this.isConnected()` is `true`) and we have pressed `Edit Credentials` (i.e. `editMode` is `true`).

Also, see that `editMode` is `false` only if we have successfully connected to a DB (i.e. if `editMode` is `false`, then `this.isConnected()` is `true`). This is the reason why I wrote first `if (!editMode`) {...}`.